src/Elements.ts

```diff
@@ -206,3 +207,194 @@ export function incoherentLightOperator(opDirPol: Operator): Operator {
   const polOutpus = Vector.fromArray([Cx(1), Cx(1)], [Dimension.polarization()])
   return opIntensity.contractLeft([1], polOutpus).contractRight([1], polInputs)
 }
+
+/**
+ * Compute operators from the grid cells
+ * @param cell, an ICell interface
+ * @returns IXYOperator
+ */
+export function generateOperator(cell: ICell): IXYOperator {
```

Comment:
This function is not too good at "don't repeat yourself".
For now, I will accept, but we need to refactor this at some point.